### PR TITLE
Require explicit dependence on `cairo_test` (i.e. `test_plugin`)"

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:83a53c59

---

**Stack**:
- #1362
- #1359 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: scarb/tests/metadata.rs [hunk 1]
```diff
@@ -143,7 +147,9 @@ fn local_dependencies() {
     assert_eq!(
         packages_and_deps(meta),
         BTreeMap::from_iter([
-            ("core".to_string(), vec!["cairo_test".to_string()]),
+            // TODO(maciektr): Update corelib manifest to include cairo_test dep.
+            // ("core".to_string(), vec!["cairo_test".to_string()]),
```
Comment (mkaput) [inline scarb/tests/metadata.rs:151]: leftover?

File: scarb/src/core/manifest/summary.rs [hunk 2]
```diff
@@ -79,27 +78,10 @@ impl Summary {
                 .version_req(DependencyVersionReq::exact(&cairo_version))
                 .build()
         });
-
-        static TEST_PLUGIN_DEPENDENCY: Lazy<ManifestDependency> = Lazy::new(|| {
```
Comment (mkaput) [inline scarb/src/core/manifest/summary.rs:83]: sounds like we can remove targets information from summaries and from registry data now?

**Review by mkaput — APPROVED** [review]

File: scarb/src/ops/resolve.rs [hunk 3]
```diff
@@ -433,6 +433,7 @@ pub struct PackageSolutionCollector<'a> {
     packages: Option<Vec<Package>>,
     cairo_plugins: Option<Vec<CompilationUnitCairoPlugin>>,
     target_kind: Option<TargetKind>,
+    warnings: HashSet<String>,
```
[the remark this answers is not in the document]
Comment (maciektr) [reply scarb/src/ops/resolve.rs:436]: Is it though? I don't really think it is 🤔

[the remark this answers is not in the document]
Comment (mkaput) [reply scarb/src/ops/resolve.rs:436]: I *believe* it could be significant when you get a cascade of warnings. But yeah, maybe I'm pedantic too much.

**Review by THenry14 — APPROVED** [review]

**Pull request merged**